Pointers/Counter.cpp

```diff
@@ -0,0 +1,30 @@
+#include "Counter.h"
+#include <iostream>
+
+using std::cout;
+using std::endl;
+
+Counter::Counter(uint32_t startValue) : m_counter(startValue) {
+	cout << "Initialize refcount" << endl;
```

Comment:
It's usually a bad idea to do I/O in the ctor, since you don't want to force output on people who might use your code in the future. I assume you added this in for debugging purposes - but it shouldn't be in the pull request. Please remove it.